Description: Upstream patch from master
 .
 cany (0.5.4-1) UNRELEASED; urgency=medium
 .
   * Import upstream release 0.5.4
Author: Jan Graichen <jg@altimos.de>

---
The information above should follow the Patch Tagging Guidelines, please
checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin: <vendor|upstream|other>, <url of original patch>
Bug: <url in upstream bugtracker>
Bug-Debian: https://bugs.debian.org/<bugnumber>
Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: <YYYY-MM-DD>

--- cany-0.5.4.orig/lib/cany/recipes/sidekiq.rb
+++ cany-0.5.4/lib/cany/recipes/sidekiq.rb
@@ -42,9 +42,9 @@ module Cany::Recipes
 
     def sidekiq_args
       args = %w(--environment production)
-      if queues.any?
+      queues.each do |queue|
         args << '--queue'
-        args << queues.join(',')
+        args << queue.to_s
       end
       args
     end
--- cany-0.5.4.orig/spec/cany/recipes/sidekiq_spec.rb
+++ cany-0.5.4/spec/cany/recipes/sidekiq_spec.rb
@@ -42,7 +42,7 @@ describe Cany::Recipes::Sidekiq do
     it 'should launch sidekiq with the list of queues' do
       expect(recipe).to receive(:install_service).with(
         :sidekiq,
-        %w(/usr/bin/test sidekiq --environment production --queue name1,name2),
+        %w(/usr/bin/test sidekiq --environment production --queue name1 --queue name2),
         user: 'www-data', group: 'www-data'
       )
       recipe.inner = double('recipe')
